r2233 | noragen | 2008-07-02 02:14:02 +0400 (Срд, 02 Июл 2008) | 2 lines

* Fix/Application of bug/patch #14117
Big thanx to sf_studio and reklov!

Backported by pva@gentoo.org.

--- plugins/icq/icqlogin.cpp	2008-07-02 16:30:39 +0000
+++ plugins/icq/icqlogin.cpp	2008-07-02 16:34:41 +0000
@@ -110,9 +110,9 @@
             m_socket->writeBuffer.tlv(0x0025, md.c_str(), md.length());
 	    if (data.owner.Uin.value){
 		m_socket->writeBuffer.tlv(0x0003, "ICQ Inc. - Product of ICQ (TM).2003b.5.56.1.3916.85");
-		m_socket->writeBuffer.tlv(0x0016, 0x010A);
-		m_socket->writeBuffer.tlv(0x0017, 0x0002);
-		m_socket->writeBuffer.tlv(0x0018, 0x0038);
+                m_socket->writeBuffer.tlv(0x0016, 0x014A); // ID Number
+                m_socket->writeBuffer.tlv(0x0017, 0x0024); // major
+                m_socket->writeBuffer.tlv(0x0018, 0x003D); // minor
 		m_socket->writeBuffer.tlv(0x0019, 0x0001);
 		m_socket->writeBuffer.tlv(0x001A, 0x0F4C);
 		m_socket->writeBuffer.tlv(0x0014, 0x00000055L);
@@ -205,9 +205,9 @@
         m_socket->writeBuffer.tlv(0x0001, uin);
         m_socket->writeBuffer.tlv(0x0002, pswd.c_str(), pswd.length());
         m_socket->writeBuffer.tlv(0x0003, "ICQ Inc. - Product of ICQ (TM).2003b.5.56.1.3916.85");
-        m_socket->writeBuffer.tlv(0x0016, 0x010A);
-        m_socket->writeBuffer.tlv(0x0017, 0x0002);
-        m_socket->writeBuffer.tlv(0x0018, 0x0038);
+	m_socket->writeBuffer.tlv(0x0016, 0x014A); // ID Number
+	m_socket->writeBuffer.tlv(0x0017, 0x0024); // major
+	m_socket->writeBuffer.tlv(0x0018, 0x003D); // minor
         m_socket->writeBuffer.tlv(0x0019, 0x0001);
         m_socket->writeBuffer.tlv(0x001A, 0x0F4C);
         m_socket->writeBuffer.tlv(0x0014, 0x00000055L);

